site/en/docs/extensions/mv3/storage-and-cookies/index.md

```diff
@@ -0,0 +1,116 @@
+pressure, although this is rare. To avoid this:
+
+- Request the `"unlimitedStorage"` permission, which affects both extension and web storage APIs and
+exempts extensions from both quota restrictions and eviction.
+- Call [`navigator.storage.persist()`][storage-persist] for protection against eviction.
+
+Extension storage is shared across the extension's origin including the extension service worker,
+any extension pages (including popups and the side panel), and offscreen documents. In content
+scripts, calling web storage APIs accesses data from the host page the content script is injected on
+and not the extension.
+
+### Access in service workers {: #storage-in-service-workers }
+
+Some web storage APIs like [IndexedDB][indexeddb] are accessible in service workers. However,
+[Local Storage][local-storage] and [Session Storage][session-storage] are not.
+
+If you have a specific use case, use an [offscreen document][offscreen] to access these.
+
+### Partitioning {: #storage-partitioning }
+
+Partitioning is where keys are introduced for stored data to limit where it can be accessed.
+
+When enabled, [storage partitioning][storage-partitioning] introduces changes to how partitioning
+keys are defined to prevent certain types of cross-site tracking. In practice, this means that if
+site A embeds an iframe containing site B, site B will not be able to access the same storage it
+would usually have when navigated to directly.
+
+To mitigate the impact of this in extensions, two key exemptions apply:
+
+- If a page with the `chrome-extension://` scheme is embedded in any site, storage partitioning will
+not apply, and the extension will have access to its top-level partition.
+- If a page with the `chrome-extension://` scheme includes an iframe, and the extension has
+[host permissions][declare-permissions] for the site it is embedding, that site will also have
+access to its top-level partition.
+
+## Cookies {: #cookies }
+
+Cookies provide a way to store key/value pairs associated with a specific domain and path. They have
+limited value in extensions but understanding their behavior can be useful if you have a specific
+use case or have bundled a third-party script which uses them in its implementation.
+
+### Secure cookies {: #secure-cookies }
+
+The [`Secure`][cookies-restrict-access] cookie attribute is only supported for the `https://`
+scheme. Consequently, `chrome-extension://` pages are not able to set cookies with this attribute.
+
+This also means that extension pages cannot use other cookie attributes where the `Secure` attribute is
+required:
+
+- [`SameSite=None`][same-site]
+- [`Partitioned`][chips]
+
+### Partitioning {: #cookies-partitioning }
+
+{% Aside %}
+See [https://crbug.com/1463991](https://crbug.com/1463991) which tracks possible changes to the
+behaviour when an extension with host permissions embeds a third-party site.
+{% endAside %}
```

Comment:
I've cc'd myself on the ticket so that we remember to update this when it's finished. Please remove this note.

Reply:
This is a bug. I was ok removing the note about the API proposal since that's future work, but I'd really like to keep this. If we make it more clear it's a bug, does that help? As some more context, as well as being long-lived documentation I expect much of the initial traffic here to be developers looking to make sure they accommodate these changes. I think it's important they know if something is not working so they can plan accordingly.

Reply:
Feel free to accept this if it looks good:

```suggestion
{% Aside %}
When an extension embeds a third-party site, that site will use the extension origin as the partition key. This means the site will not be able access the same cookies as if it was navigated to directly. See [https://crbug.com/1463991](https://crbug.com/1463991).
{% endAside %}
```